pkitesting/src/main/java/io/netty/pkitesting/Algorithms.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.pkitesting;
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+import java.security.GeneralSecurityException;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.security.SecureRandom;
+import java.security.Signature;
+import java.security.spec.AlgorithmParameterSpec;
+
+final class Algorithms {
+    private static Provider bouncyCastle;
+
+    private Algorithms() {
+    }
+
+    static String oidForAlgorithmName(String algorithmIdentifier) {
+        // See the Java Security Standard Algorithm Names documentation for names and links to RFCs.
+        // https://docs.oracle.com/en/java/javase/22/docs/specs/security/standard-names.html#signature-algorithms
+        switch (algorithmIdentifier) {
+            case "SHA256withECDSA":
```

Comment:
I know these naming conventions are standard but if there is case mismatch, this will break apart. Perhaps, `toLowerCase` everything?